ExportKindleClippingsToNotion/ClippingsExtractor.cs

```diff
@@ -0,0 +1,63 @@
+using System.Text.RegularExpressions;
+
+namespace ExportKindleClippingsToNotion;
+
+public class ClippingsExtractor
+{
+    public IReadOnlyList<Clipping> ExtractClippings(string file)
+    {
+        var clippingSections = file
+            .Split("==========")
+            .Where(clipping => clipping.Trim().Length > 0);
+
+        var clippings = new List<Clipping>();
+
+        foreach (var clippingSection in clippingSections)
+        {
+            var lines = clippingSection
+                .Split("\r\n")
+                .Where(line => line.Length > 0)
+                .Select(line => line.Trim())
+                .Select(line => line.Replace("\u200B", string.Empty))
+                .Select(line => line.Replace("\uFEFF", string.Empty))
+                .ToList();
+
+            if (lines.Count != 3)
+                continue;
+
+            var firstLineRegex = new Regex(@"^(?<Title>.*?)(\((?<Year>\d{4})\) )?\((?<AuthorName>[^\(]*?)\)$");
```

Comment:
Ich würde den Regex und die Gruppen in Konstanten auslagern. Ist einfach besser lesbarer und weniger magic